src/main/java/bio/terra/service/snapshot/flight/create/CreateSnapshotValidateQueryStep.java

```diff
@@ -35,6 +31,7 @@ public StepResult doStep(FlightContext context) {
      * make sure the user has custodian data access (currently this is done in the controller,
      * but this should be moved
      */
+    SnapshotRequestModel snapshotReq = getByQueryRequestModel(context);
```

Comment:
Can you remind me why this shared method is needed? It seems like it is behaving the same as just passing the snapshot request in from the flight.

Reply:
Yes, that's true it is behaving the same. We need this if we are going to store the converted byRequestId (now ByQuery) in the map. It can't be passed into the step, it needs to be retrieved from the map. So this change was made to accommodate the new case. But the way that it is currently implemented, creating a query and an asset and not a full ByQuery model, we don't need this.

Reply:
I've decided to revert this change